packages/shared/src/components/ProfilePicture.tsx

```diff
@@ -102,21 +130,30 @@ function ProfilePictureComponent(
     return null;
   }
 
+  const isImageResizable =
+    user.image &&
+    [
+      ProfileImageSize.XLarge,
+      ProfileImageSize.Large,
+      ProfileImageSize.Medium,
+      ProfileImageSize.Small,
+      ProfileImageSize.XSmall,
+    ].includes(size);
   const imageAlt = `${user.username || user.name || user.id}'s profile`;
+  const imageSrc = isImageResizable ? getResizedSrc(user.image) : user.image;
```

Comment:
Also maybe memo together with https://github.com/dailydotdev/apps/pull/3679/files#r1816451466, since regex matching is also involved and would be repeated on every render.

Reply:
Yes! Updated with useMemo, thanks for suggestion!